### DISCUSSION TITLE
Spam extensions on the Marketplace

### DISCUSSION THREAD

**Discussion by prashantvc** [microsoft/vsmarketplace#343] [Announcement]
We are seeing high number of spam extensions in the marketplace. The team is actively working on preventing these spam extensions.

Meantime you may experience service degradation

We appreciate your patience ❤️

**Comment by zimmski** [comment]
Can you please elaborate on what a "spam extension" is? We opened #346 because the README did not change. We have publishing a new version in our CI. So it can happen that the README is changed multiple times a day. Is that maybe spam already?

**Reply by ssbarnea** [reply]
There is another issue related to nature-spamming (not deliberate one). As vast majority of extensions are released by **individuals**, from theirs own github accounts it is likely that after a number of months or years they become abandonware.

As most of them are open source, forking, changing the name and releasing again is a common practice. I even seen caused with a chain of 8-10 extensions.

vscode-marketplace team has it own contribution on this because the ability to mark an extension as abandoned was never implemented. The focus was to brag about how many extensions there are. The reality is that **more is less** in that case. All leftovers do devalue the entire marketplace as it makes much harder to spot the active ones.

Microsoft could persuade people to join efforts when developing extensions, like promoting extensions that do come from organizations (github ones, not corporations) and that have more than one maintainer. Any project that has more than one maintainer is far more likely to resist the test of time. You don't really need to organize people for this but setting some guidelines and distinguishing between one-man-extension and community backed ones could be enough to prevent explosion of forked extensions.

I based my statement on what happened related to Ansible extensions, where I spent the last 1+ year contacting authors or older extensions and convincing them to team-up and work on maintaining a single Ansible extension. It required some time but I am pleased about the result as most of these outdated extensions were unpublished by now.

A certification program could be another way to distinguish between low and high quality extensions but that is clearly something that needs considerably more effort.

PS. I happened that few months back I wrote, over a weekend,  an extension that does only uninstall known outdated or broken extensions: https://marketplace.visualstudio.com/items?itemName=42picky.octarine -- I was about to call it the "uninstaller" but decided to go for something else, in case other features will be added after.

**Comment by hl2guide** [comment]
@prashantvc Please continue to clean this up.

Some examples I've noticed today:

_Chinese text spam:_

![Screenshot 2022-03-14 040630](https://user-images.githubusercontent.com/17044982/158070872-16277bce-9bfa-4571-9f35-0579e0415dd5.jpg)

_Robux spam:_

![Screenshot 2022-03-14 040654](https://user-images.githubusercontent.com/17044982/158070874-c79858c1-627e-4513-a4ea-52bbf83aad0d.jpg)

_More Chinese text spam:_

![Screenshot 2022-03-14 040712](https://user-images.githubusercontent.com/17044982/158070876-3eed5472-721e-479c-b348-b955f475e26d.jpg)

**Comment by hl2guide** [comment]
some more spam extensions:

![1](https://user-images.githubusercontent.com/17044982/160293139-26db7b8d-252d-408a-a41a-37f60c1fb160.jpg)
![2](https://user-images.githubusercontent.com/17044982/160293140-735182d5-c366-4722-a10f-04bb2001bc9f.jpg)

Seems like the spam posters are just using keywords that a currently popular (Oscars, Tszyu).

**Comment by hl2guide** [comment]
@prashantvc Please continue to work on this on-going problem (as of April 2022):

_Suspected spam/scam extensions are marked in yellow below_

![Screenshot 2022-04-30 132411](https://user-images.githubusercontent.com/17044982/166089041-36f9c55e-017f-4573-8e9a-e747b9ddaaf6.jpg)
![Screenshot 2022-04-30 132507](https://user-images.githubusercontent.com/17044982/166089042-b8b81d40-cb2c-4af4-b094-4321476ab736.jpg)
![Screenshot 2022-04-30 132600](https://user-images.githubusercontent.com/17044982/166089043-8313b62b-d18c-4e30-bae9-f0daa1d39951.jpg)
![Screenshot 2022-04-30 132629](https://user-images.githubusercontent.com/17044982/166089044-549f09f5-d2b9-4851-ab1c-f14d09035102.jpg)
![Screenshot 2022-04-30 132835](https://user-images.githubusercontent.com/17044982/166089046-8c845eb4-cb6d-48c7-80af-de0e70066f41.jpg)

**Comment by hl2guide** [comment]
More:
![Screenshot 2022-05-01 143339](https://user-images.githubusercontent.com/17044982/166132359-67f94343-ccd3-4082-815e-e4ac3b12045e.jpg)
